integration-tests/README.md

```diff
@@ -98,3 +104,29 @@ By Default test runs with those authentication parameters for Keycloak:
 ```
 
 You can update the parameters needs to configure in `integration-tests/pom.xml` or via the maven command.
+
+### Performance testing
+
+In order to be able to run the [performance tests scenarios](src/test/resources/performance-features) locally you will need to have the following installed on your machine:
+
+- [Hyperfoil v0.21](https://hyperfoil.io/) - Benchmark framework for microservices (*)
+- [webhook-perf-test](https://github.com/afalhambra/webhook-perf-test) - Dummy app to consume cloud events from webhook Processor
+
+There is a dedicated Maven profile named `performance` located in `integration-tests/pom.xml`. There you can see the following system properties needed 
+to run these performance test scenarios:
+
+```xml
+<slack.webhook.performance.url></slack.webhook.performance.url>
+<performance.hyperfoil.url></performance.hyperfoil.url>
+```
+
+Once you have `Hyperfoil` and the `webhook-perf-test` up and running you can run the performance tests by either replacing the properties mentioned above in the pom
+file or by passing them as an argument via a Maven command. For example:
+```bash
+mvn clean verify -Pperformance -Dslack.webhook.performance.url=<WEBHOOK_PERF_URL> -Dperformance.hyperfoil.url=<HYPERFOIL_URL> -Devent-bridge.manager.url=<MANAGER_URL> -Dkeycloak.realm.url=<KEYCLOAK_URL> -Dwebhook.site.uuid=<WEBHOOK_SITE_UUID>
```

Comment:
Do we need also `webhook.site.uuid` ? It is used  for non performance tests of webhook sink processor. I am concerned using the same for performance will overload it and cause other tests to fail.

Reply:
Yes, we need to define that property when running the performance test, that's because the [Hook class](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/Hooks.java) is used across all the cucumber tests and cannot be split as far as I know. [In there](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/Hooks.java#L65), you can see that a request to the WebhookSiteResource is done, so therefore this property is required

These performance tests will only be executed when profile is set to `performance` - not enabled by default.

Reply:
@jomarko I've just made some changes to the code, so that property `webhook.site.uuid` is no longer required when running the performance tests. Please have a look

Reply:
@afalhambra thank you for answer. then I think. We should update the method slightly.

from
```java
    @BeforeAll(order = 1)
    public static void webhookSiteRequestHistoryIsCleared() {
        final LocalDate yesterday = LocalDate.now(ZoneId.systemDefault()).minusDays(1);
        WebhookSiteResource.requests(WebhookSiteQuerySorting.OLDEST)
                .stream()
                .filter(request -> {
                    final LocalDate requestCreatedAt = request.getCreatedAt().toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
                    return yesterday.isAfter(requestCreatedAt);
                })
                .forEach(request -> WebhookSiteResource.deleteRequest(request));
    }
```    
to    
```java
    @BeforeAll(order = 1)
    public static void webhookSiteRequestHistoryIsCleared() {
        if (!WebhookPerformanceResource.isSpecified()) {
            final LocalDate yesterday = LocalDate.now(ZoneId.systemDefault()).minusDays(1);
            WebhookSiteResource.requests(WebhookSiteQuerySorting.OLDEST)
                    .stream()
                    .filter(request -> {
                        final LocalDate requestCreatedAt = request.getCreatedAt().toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
                        return yesterday.isAfter(requestCreatedAt);
                    })
                    .forEach(request -> WebhookSiteResource.deleteRequest(request));
        }
    }
```

My understanding is `webhookSiteRequestHistoryIsCleared` runs always, both for performance and non performance profile. My concern is then, that if some job with `performance` profile is started a in a middle of non performance job execution, webhook.site may be cleared and non performance job may fail. I understand the chance is small, as `webhookSiteRequestHistoryIsCleared` does not clear everything.

However adding this `if` statement wouldn't require `webhook.site.uuid` to be set for performance jobs, or am I wrong? What is your opinion?

Reply:
I've just changed slightly the code so now `webhookSiteRequestHistoryIsCleared` won't be executed when running performance tests (it is not required), so that `webhook.site.uuid` is no longer needed for `performance` tests